{"uid":"46c75da15dbb70b","name":"TC 01_15_01 Verify possibility to vote with 3 stars","fullName":"tests.test_reviews_page.TestReviews#test_tc_01_15_03_check_that_it_is_possible_to_vote_for_3_star","historyId":"a09b3b388c7a950ae742890d45064f2f","time":{"start":1718156575485,"stop":1718156580689,"duration":5204},"description":"\n        The user is NOT logged in, located in the product card, section \"Reviews\"\n        1. All required fields are filled with correct data\n        2. Pressed 3 star\n        3.The \"Submit Review\" button is pressed\n        Result: Present in the DOM and visible on the page the message: \"You submitted your review for moderation.\n        This code works when run on Win, Mac, Ubuntu on a real laptop and has the features of working on GitHub Action.\n        When interacting with the GitHub Action, it checks not for the text of the \"Success\" message, but for the \"Success\" URL\"\n        ","descriptionHtml":"<pre><code>    The user is NOT logged in, located in the product card, section &quot;Reviews&quot;\n    1. All required fields are filled with correct data\n    2. Pressed 3 star\n    3.The &quot;Submit Review&quot; button is pressed\n    Result: Present in the DOM and visible on the page the message: &quot;You submitted your review for moderation.\n    This code works when run on Win, Mac, Ubuntu on a real laptop and has the features of working on GitHub Action.\n    When interacting with the GitHub Action, it checks not for the text of the &quot;Success&quot; message, but for the &quot;Success&quot; URL&quot;\n</code></pre>\n","status":"skipped","statusMessage":"XFAIL Flaky behavior on CI\n\nselenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55f88681fe3a <unknown>\n#1 0x55f88650945c <unknown>\n#2 0x55f8865555b5 <unknown>\n#3 0x55f886555671 <unknown>\n#4 0x55f886599f14 <unknown>\n#5 0x55f8865784dd <unknown>\n#6 0x55f8865972cc <unknown>\n#7 0x55f886578253 <unknown>\n#8 0x55f8865481c7 <unknown>\n#9 0x55f886548b3e <unknown>\n#10 0x55f8867e627b <unknown>\n#11 0x55f8867ea327 <unknown>\n#12 0x55f8867d2dae <unknown>\n#13 0x55f8867eadf2 <unknown>\n#14 0x55f8867b774f <unknown>\n#15 0x55f88680f128 <unknown>\n#16 0x55f88680f2fb <unknown>\n#17 0x55f88681ef6c <unknown>\n#18 0x7f1704094ac3 <unknown>","statusTrace":"self = <test_reviews_page.TestReviews object at 0x7f958bef9ed0>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"537f9a4e876daa0b32fee7c9abf23a79\")>\n\n    @pytest.mark.xfail(reason=\"Flaky behavior on CI\")\n    @allure.title('TC 01_15_01 Verify possibility to vote with 3 stars')\n    def test_tc_01_15_03_check_that_it_is_possible_to_vote_for_3_star(self, driver):\n        \"\"\"\n        The user is NOT logged in, located in the product card, section \"Reviews\"\n        1. All required fields are filled with correct data\n        2. Pressed 3 star\n        3.The \"Submit Review\" button is pressed\n        Result: Present in the DOM and visible on the page the message: \"You submitted your review for moderation.\n        This code works when run on Win, Mac, Ubuntu on a real laptop and has the features of working on GitHub Action.\n        When interacting with the GitHub Action, it checks not for the text of the \"Success\" message, but for the \"Success\" URL\"\n        \"\"\"\n    \n        \"\"\"Steps\"\"\"\n        page = ReviewsPage(driver, REVIEWS_URL_GENERAL)\n        page.open()\n    \n        \"\"\"Get the current URL and print it\"\"\"\n        page.get_the_current_URL_and_print_it()\n    \n        \"\"\"Steps\"\"\"\n>       page.open_review_menu()\n\ntests/test_reviews_page.py:146: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/reviews_page.py:14: in open_review_menu\n    review_menu = self.element_is_clickable(self.locators.REVIEW_MENU)\npages/base_page.py:80: in element_is_clickable\n    return wait(self.driver, timeout).until(EC.element_to_be_clickable(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"537f9a4e876daa0b32fee7c9abf23a79\")>\nmethod = <function element_to_be_clickable.<locals>._predicate at 0x7f958c16a8e0>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x55f88681fe3a <unknown>\nE       #1 0x55f88650945c <unknown>\nE       #2 0x55f8865555b5 <unknown>\nE       #3 0x55f886555671 <unknown>\nE       #4 0x55f886599f14 <unknown>\nE       #5 0x55f8865784dd <unknown>\nE       #6 0x55f8865972cc <unknown>\nE       #7 0x55f886578253 <unknown>\nE       #8 0x55f8865481c7 <unknown>\nE       #9 0x55f886548b3e <unknown>\nE       #10 0x55f8867e627b <unknown>\nE       #11 0x55f8867ea327 <unknown>\nE       #12 0x55f8867d2dae <unknown>\nE       #13 0x55f8867eadf2 <unknown>\nE       #14 0x55f8867b774f <unknown>\nE       #15 0x55f88680f128 <unknown>\nE       #16 0x55f88680f2fb <unknown>\nE       #17 0x55f88681ef6c <unknown>\nE       #18 0x7f1704094ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"_session_faker","time":{"start":1718155045618,"stop":1718155045680,"duration":62},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"clear_allure_result_folder","time":{"start":1718155045680,"stop":1718155045680,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"driver","time":{"start":1718156574213,"stop":1718156575484,"duration":1271},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"\n        The user is NOT logged in, located in the product card, section \"Reviews\"\n        1. All required fields are filled with correct data\n        2. Pressed 3 star\n        3.The \"Submit Review\" button is pressed\n        Result: Present in the DOM and visible on the page the message: \"You submitted your review for moderation.\n        This code works when run on Win, Mac, Ubuntu on a real laptop and has the features of working on GitHub Action.\n        When interacting with the GitHub Action, it checks not for the text of the \"Success\" message, but for the \"Success\" URL\"\n        ","status":"skipped","statusMessage":"XFAIL Flaky behavior on CI\n\nselenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55f88681fe3a <unknown>\n#1 0x55f88650945c <unknown>\n#2 0x55f8865555b5 <unknown>\n#3 0x55f886555671 <unknown>\n#4 0x55f886599f14 <unknown>\n#5 0x55f8865784dd <unknown>\n#6 0x55f8865972cc <unknown>\n#7 0x55f886578253 <unknown>\n#8 0x55f8865481c7 <unknown>\n#9 0x55f886548b3e <unknown>\n#10 0x55f8867e627b <unknown>\n#11 0x55f8867ea327 <unknown>\n#12 0x55f8867d2dae <unknown>\n#13 0x55f8867eadf2 <unknown>\n#14 0x55f8867b774f <unknown>\n#15 0x55f88680f128 <unknown>\n#16 0x55f88680f2fb <unknown>\n#17 0x55f88681ef6c <unknown>\n#18 0x7f1704094ac3 <unknown>","statusTrace":"self = <test_reviews_page.TestReviews object at 0x7f958bef9ed0>\ndriver = <selenium.webdriver.chrome.webdriver.WebDriver (session=\"537f9a4e876daa0b32fee7c9abf23a79\")>\n\n    @pytest.mark.xfail(reason=\"Flaky behavior on CI\")\n    @allure.title('TC 01_15_01 Verify possibility to vote with 3 stars')\n    def test_tc_01_15_03_check_that_it_is_possible_to_vote_for_3_star(self, driver):\n        \"\"\"\n        The user is NOT logged in, located in the product card, section \"Reviews\"\n        1. All required fields are filled with correct data\n        2. Pressed 3 star\n        3.The \"Submit Review\" button is pressed\n        Result: Present in the DOM and visible on the page the message: \"You submitted your review for moderation.\n        This code works when run on Win, Mac, Ubuntu on a real laptop and has the features of working on GitHub Action.\n        When interacting with the GitHub Action, it checks not for the text of the \"Success\" message, but for the \"Success\" URL\"\n        \"\"\"\n    \n        \"\"\"Steps\"\"\"\n        page = ReviewsPage(driver, REVIEWS_URL_GENERAL)\n        page.open()\n    \n        \"\"\"Get the current URL and print it\"\"\"\n        page.get_the_current_URL_and_print_it()\n    \n        \"\"\"Steps\"\"\"\n>       page.open_review_menu()\n\ntests/test_reviews_page.py:146: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npages/reviews_page.py:14: in open_review_menu\n    review_menu = self.element_is_clickable(self.locators.REVIEW_MENU)\npages/base_page.py:80: in element_is_clickable\n    return wait(self.driver, timeout).until(EC.element_to_be_clickable(locator))\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"537f9a4e876daa0b32fee7c9abf23a79\")>\nmethod = <function element_to_be_clickable.<locals>._predicate at 0x7f958c16a8e0>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x55f88681fe3a <unknown>\nE       #1 0x55f88650945c <unknown>\nE       #2 0x55f8865555b5 <unknown>\nE       #3 0x55f886555671 <unknown>\nE       #4 0x55f886599f14 <unknown>\nE       #5 0x55f8865784dd <unknown>\nE       #6 0x55f8865972cc <unknown>\nE       #7 0x55f886578253 <unknown>\nE       #8 0x55f8865481c7 <unknown>\nE       #9 0x55f886548b3e <unknown>\nE       #10 0x55f8867e627b <unknown>\nE       #11 0x55f8867ea327 <unknown>\nE       #12 0x55f8867d2dae <unknown>\nE       #13 0x55f8867eadf2 <unknown>\nE       #14 0x55f8867b774f <unknown>\nE       #15 0x55f88680f128 <unknown>\nE       #16 0x55f88680f2fb <unknown>\nE       #17 0x55f88681ef6c <unknown>\nE       #18 0x7f1704094ac3 <unknown>\n\n/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException","steps":[{"name":"Open a browser","time":{"start":1718156575485,"stop":1718156575572,"duration":87},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"Find clickable elements","time":{"start":1718156575576,"stop":1718156580688,"duration":5112},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55f88681fe3a <unknown>\n#1 0x55f88650945c <unknown>\n#2 0x55f8865555b5 <unknown>\n#3 0x55f886555671 <unknown>\n#4 0x55f886599f14 <unknown>\n#5 0x55f8865784dd <unknown>\n#6 0x55f8865972cc <unknown>\n#7 0x55f886578253 <unknown>\n#8 0x55f8865481c7 <unknown>\n#9 0x55f886548b3e <unknown>\n#10 0x55f8867e627b <unknown>\n#11 0x55f8867ea327 <unknown>\n#12 0x55f8867d2dae <unknown>\n#13 0x55f8867eadf2 <unknown>\n#14 0x55f8867b774f <unknown>\n#15 0x55f88680f128 <unknown>\n#16 0x55f88680f2fb <unknown>\n#17 0x55f88681ef6c <unknown>\n#18 0x7f1704094ac3 <unknown>\n\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/runner/work/luma_project/luma_project/pages/base_page.py\", line 80, in element_is_clickable\n    return wait(self.driver, timeout).until(EC.element_to_be_clickable(locator))\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/opt/hostedtoolcache/Python/3.11.9/x64/lib/python3.11/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[{"name":"locator","value":"('css selector', '#tab-label-reviews-title')"},{"name":"timeout","value":"5"}],"shouldDisplayMessage":true,"stepsCount":0,"attachmentsCount":0,"hasContent":true,"attachmentStep":false}],"attachments":[{"uid":"8b04bf761878b28c","name":"Screenshot","source":"8b04bf761878b28c.png","type":"image/png","size":25664},{"uid":"302651ef0c19d23f","name":"HTML source","source":"302651ef0c19d23f.html","type":"text/html","size":774}],"parameters":[],"shouldDisplayMessage":true,"stepsCount":2,"attachmentsCount":2,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"driver::0","time":{"start":1718156580800,"stop":1718156580865,"duration":65},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"clear_allure_result_folder::0","time":{"start":1718157437577,"stop":1718157437577,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"shouldDisplayMessage":false,"stepsCount":0,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"epic","value":"Review Page"},{"name":"tag","value":"@pytest.mark.xfail(reason='Flaky behavior on CI')"},{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_reviews_page"},{"name":"subSuite","value":"TestReviews"},{"name":"host","value":"fv-az802-720"},{"name":"thread","value":"2834-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_reviews_page"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[],"history":{"statistic":{"failed":0,"broken":0,"skipped":1,"passed":3,"unknown":0,"total":4},"items":[{"uid":"1c7025f81e9e1143","reportUrl":"https://verafes.github.io/luma_project/338//#testresult/1c7025f81e9e1143","status":"passed","statusDetails":"XPASS Flaky behavior on CI","time":{"start":1718069421721,"stop":1718069435299,"duration":13578}},{"uid":"cf7c799baa08e5bc","reportUrl":"https://verafes.github.io/luma_project/337//#testresult/cf7c799baa08e5bc","status":"passed","statusDetails":"XPASS Flaky behavior on CI","time":{"start":1717983144065,"stop":1717983157554,"duration":13489}},{"uid":"80b1e8a35fe3c30f","reportUrl":"https://verafes.github.io/luma_project/336//#testresult/80b1e8a35fe3c30f","status":"passed","statusDetails":"XPASS Flaky behavior on CI","time":{"start":1717897050636,"stop":1717897063117,"duration":12481}}]},"tags":["@pytest.mark.xfail(reason='Flaky behavior on CI')"]},"source":"46c75da15dbb70b.json","parameterValues":[]}